apps/qubit/modules/informationobject/actions/deleteFindingAidAction.class.php

```diff
@@ -17,24 +17,39 @@
  * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
  */
 
-class InformationObjectFindingAidAction extends sfAction
+class InformationObjectDeleteFindingAidAction extends sfAction
 {
-  // export an object w/relations as an XML document with selected schema
   public function execute($request)
   {
-    if (isset($this->getRoute()->resource) && sfContext::getInstance()->user->isAuthenticated())
+    $this->resource = $this->getRoute()->resource;
+
+    // Check that this isn't the root
+    if (!isset($this->resource->parent))
+    {
+      $this->forward404();
+    }
+
+    // Check user authorization
+    if (!$this->context->user->isAuthenticated())
+    {
+      QubitAcl::forwardUnauthorized();
+    }
+
+    $this->form = new sfForm;
+
+    if ($request->isMethod('delete'))
     {
       $i18n = $this->context->i18n;
-      $resource = $this->getRoute()->resource;
 
       $params = array(
-        'objectId' => $resource->id,
-        'description' => $i18n->__('Generating finding aid for: ').$resource->getTitle(array('cultureFallback' => true))
+        'objectId'    => $this->resource->id,
+        'description' => $i18n->__('Deleting finding aid for: ').$this->resource->getTitle(array('cultureFallback' => true)),
+        'delete'      => true
       );
 
-      $job = QubitJob::runJob('arGenerateFindingAidJob', $params);
-    }
+      $job = QubitJob::runJob('arFindingAidJob', $params);
```

Comment:
What do you use `$job` for? I guess because this is running asynchronously you can just fire and forget? But if it's posible to receive some kind of error you should probably check it.